pkg/controller/statusmanager/status_manager.go

```diff
@@ -113,9 +114,11 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...confi
 			log.Printf("Created ClusterOperator with conditions:\n%s", string(buf))
 		}
 	} else {
-		err = status.client.Status().Update(context.TODO(), co)
-		if err != nil {
-			log.Printf("Failed to update ClusterOperator %q: %v", co.Name, err)
+		resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
Is this going to work? I don't think so. You need to re-fetch `co` and re-update it.

Reply:
> Is this going to work? I don't think so.

Why is that?

Reply:
https://github.com/kubernetes/kubernetes/pull/82284/files

Reply:
If that doesn't answer your question then let me know so I can make further improvements there...

Reply:
That makes it more clear. Thanks for that PR